types/module/server/router.go

```diff
@@ -68,71 +85,71 @@ func (r registrar) RegisterService(sd *grpc.ServiceDesc, ss interface{}) {
 }
 
 func (rtr *router) invoker(methodName string, writeCondition func(context.Context, string, sdk.Msg) error) (types.Invoker, error) {
-	handler, found := rtr.handlers[methodName]
-	if !found {
-		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("cannot find method named %s", methodName))
-	}
-
-	moduleName := handler.moduleName
-
-	if writeCondition != nil && handler.commitWrites {
-		// msg handler
-		return func(ctx context.Context, request interface{}, response interface{}, opts ...interface{}) error {
-			if rtr.antiReentryMap[moduleName] {
-				return fmt.Errorf("re-entrant module calls not allowed for security reasons! module %s is already on the call stack", moduleName)
-			}
+	return func(ctx context.Context, request interface{}, response interface{}, opts ...interface{}) error {
+		req, ok := request.(proto.Message)
+		if !ok {
+			return fmt.Errorf("expected proto.Message, got %T for service method %s", request, methodName)
+		}
+		handler, found := rtr.handlers[TypeURL(req)]
 
-			rtr.antiReentryMap[moduleName] = true
-			defer delete(rtr.antiReentryMap, moduleName)
+		msg, isMsg := request.(sdk.Msg)
+		legacyMsg, isLegacyMsg := request.(legacytx.LegacyMsg)
+		if !found && !isLegacyMsg {
+			return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("cannot find method named %s", methodName))
```

Comment:
I have a question here: does `invoker` here also handle inter-module queries? I believe in this PR, we only use this for inter-module Msgs.

But if yes, iiuc this would fail. For example if we use ADR-033 to query `cosmos.bank.v1beta1.Balances`, then `found == false` because bank is not registered in this ADR-033 router, and `isLegacyMsg == false`.

Reply:
yeah it also handles inter-module queries but for it to work, they would have to be registered using the ADR 033 router anyway